clippy_lints/src/methods.rs

```diff
@@ -964,6 +992,96 @@ fn lint_or_fun_call(cx: &LateContext, expr: &hir::Expr, method_span: Span, name:
     }
 }
 
+/// Checks for the `EXPECT_FUN_CALL` lint.
+fn lint_expect_fun_call(cx: &LateContext, expr: &hir::Expr, method_span: Span, name: &str, args: &[hir::Expr]) {
+    fn check_general_case(
+        cx: &LateContext,
+        name: &str,
+        method_span: Span,
+        self_expr: &hir::Expr,
+        arg: &hir::Expr,
+        span: Span,
+    ) {
+        if name != "expect" {
+            return;
+        }
+
+        let self_type = cx.tables.expr_ty(self_expr);
+        let known_types = &[&paths::OPTION, &paths::RESULT];
+
+        // if not a known type, return early
+        if known_types.iter().all(|&k| !match_type(cx, self_type, k)) {
+            return;
+        }
+
+        // don't lint for constant values
+        let owner_def = cx.tcx.hir.get_parent_did(arg.id);
+        let promotable = cx.tcx.rvalue_promotable_map(owner_def).contains(&arg.hir_id.local_id);
+        if promotable {
+            return;
+        }
+
+        let closure = if match_type(cx, self_type, &paths::OPTION) { "||" } else { "|_|" };
+        let span_replace_word = method_span.with_hi(span.hi());
+
+        if let hir::ExprAddrOf(_, ref addr_of) = arg.node {
+            if let hir::ExprCall(ref _inner_fun, ref inner_args) = addr_of.node {
+                // TODO: check if inner_fun is call to format!
```

Comment:
@oli-obk also here, I wasn't sure if there was a way to easily check the function call in a similar way to how the `paths.rs` file constants can be used to check types. Again, any pointers would be appreciated.

Reply:
We do have a check for asking which macro you're in. try grepping for `"unimplemented"` or `"panic"`